test/style/source.test.cpp

```diff
@@ -708,6 +708,75 @@ TEST(Source, GeoJSonSourceUrlUpdate) {
     test.run();
 }
 
+TEST(Source, GeoJSONSourceSupercluster) {
+    SourceTest test;
+
+    test.fileSource.sourceResponse = [&] (const Resource& resource) {
+        EXPECT_EQ("http://url", resource.url);
+        Response response;
+        response.data = std::make_unique<std::string>(util::read_file("test/fixtures/supercluster/places.json"));
+        return response;
+    };
+
+    test.styleObserver.sourceLoaded = [&](Source& source) {
+        EXPECT_TRUE(source.is<GeoJSONSource>());
+        const auto geojson_source = source.as<GeoJSONSource>();
+        ASSERT_NE(geojson_source, nullptr);
+        EXPECT_FALSE(geojson_source->getClusterChildren(1).empty());
+        EXPECT_FALSE(geojson_source->getClusterLeaves(1).empty());
+        EXPECT_NE(geojson_source->getClusterExpansionZoom(1), 0);
```

Comment:
Can we test that the API returns specific values here instead of just making sure they're not default ones like for non-clustered sources?

Reply:
@mourner Done. I noticed that default [GeoJSONSource](https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/style/sources/geojson_source.hpp#L25) clustering options differ from supercluster's defaults. Should default options be synced or there was a specific reason for  different default options?

Reply:
Supercluster (the JS version) can be used in different contexts than GL Native (e.g. with Leaflet), so it's not a problem that we don't match default options.